core/src/main/java/feast/core/model/FeatureTable.java

```diff
@@ -0,0 +1,323 @@
+ */
+package feast.core.model;
+
+import com.google.protobuf.Duration;
+import com.google.protobuf.Timestamp;
+import feast.core.dao.EntityRepository;
+import feast.core.util.TypeConversion;
+import feast.proto.core.FeatureProto.FeatureSpecV2;
+import feast.proto.core.FeatureSourceProto.FeatureSourceSpec;
+import feast.proto.core.FeatureTableProto;
+import feast.proto.core.FeatureTableProto.FeatureTableSpec;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Entity
+@Setter(AccessLevel.PRIVATE)
+@Table(
+    name = "feature_tables",
+    uniqueConstraints = @UniqueConstraint(columnNames = {"name", "project_name"}))
+public class FeatureTable extends AbstractTimestampEntity {
+
+  @Id @GeneratedValue private long id;
+
+  // Name of Feature Table
+  @Column(name = "name", nullable = false)
+  private String name;
+
+  // Name of the Project that this FeatureTable belongs to
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "project_name")
+  private Project project;
+
+  // Features defined in this Feature Table
+  @OneToMany(
+      mappedBy = "featureTable",
+      cascade = CascadeType.ALL,
+      fetch = FetchType.EAGER,
+      orphanRemoval = true)
+  private Set<FeatureV2> features;
+
+  // Entites to associate the features defined in this FeatureTable with
+  @ManyToMany
+  @JoinTable(
+      name = "feature_tables_entities_v2",
+      joinColumns = @JoinColumn(name = "feature_table_id"),
+      inverseJoinColumns = @JoinColumn(name = "entity_v2_id"))
+  private Set<EntityV2> entities;
+
+  // User defined metadata labels serialized as JSON string.
+  @Column(name = "labels", columnDefinition = "text")
+  private String labelsJSON;
+
+  // Max Age of the Features defined in this Feature Table in seconds
+  @Column(name = "max_age", nullable = false)
+  private long maxAgeSecs;
+
+  // Streaming Feature Source used to obtain data for features from a stream
+  @OneToOne(cascade = CascadeType.ALL)
+  @JoinColumn(name = "stream_source_id", nullable = true)
+  private FeatureSource streamSource;
+
+  // Batched Feature Source used to obtain data for features from a batch of data
+  @OneToOne(cascade = CascadeType.ALL)
+  @JoinColumn(name = "batch_source_id", nullable = true)
```

Comment:
Why is this nullable?

Reply:
Updated to make batch source compulsory.